src/view/components/common/iframe-legacy/Iframe-legacy-component.ts

```diff
@@ -65,4 +66,22 @@ export class IframeLegacyComponent extends SiteChangeListener {
             this.iframeElement.contentWindow.location.reload();
         }
     }
+
+    /**
+     * Validate if the iframe window is send to the login page after jsessionid expired
+     * then logout the user from angular session
+     */
+    checkSessionExpired(): void {
+        if (this.iframeElement && this.iframeElement.contentWindow) {
+            let currentPath = this.iframeElement.contentWindow.location.pathname;
+
+            if (currentPath.indexOf('/c/portal_public/login') != -1) {
```

Comment:
Unfortunately, this is not easy to accomplish because the variable indicating that we are using an iframe is pass only one time as a request parameter, after that is saved in session, so if the user is navigating inside the iframe the parameter is kept in session but at this point, the session doesn't exist anymore